.github/workflows/tests.yml

```diff
@@ -0,0 +1,101 @@
+---
+name: Terraform modules tests
+
+on:
+  schedule:
+    - cron: '0 1 * * 2'
+  workflow_dispatch:
+  pull_request:
+
+
+env:
+  AWS_PROFILE: "infex"
+  AWS_REGION: "eu-central-1"
+
+jobs:
+  integration-test:
+    runs-on: ubuntu-22.04
+    # Other dependencies from hosted runner
+    # AWS CLI
+    # indirectly node used by actions
+
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4
+
+      - name: Check labels
+        run: |
+          labels=$(jq -r '.pull_request.labels[].name' "$GITHUB_EVENT_PATH")
+          echo "Labels: $labels"
+          if ! echo "$labels" | grep -qE "^testing-allowed$|^testing-ci-not-necessary$"; then
```

Comment:
here and following lines.
I don't understand the purpose of `testing-ci-not-necessary`.

Reply:
sounds to me like that.
Even if it's auto labeled via a different workflow, it will likely be too slow for this workflow to catch it.

Reply:
I still don't understand why we need it in the first place.

Reply:
if you want to keep it, then I'd go with a solution like here: https://github.com/camunda/web-modeler/blob/main/.github/workflows/preview-env-teardown.yml#L10
It's a bit cleaner on a job level than every single step.

Also probably a single label is more than sufficient to decide on whether to trigger or not.
Additionally, one could add an exception for Renovate (`GITHUB_ACTOR`)

Reply:
second thought on the opinion, in previous companies we just had a label called `skip-ci` that was a universal way to claim should be skipped in the test - maybe we can just use this one label here, and do the filtering on the job level (e.g. if the label `skip-ci` is present -> don't run the terratest checks)

Reply:
or just simply rely on the built-in capabilities --> https://docs.github.com/en/actions/managing-workflow-runs/skipping-workflow-runs